### PR TITLE
Onboard endnode

### PR DESCRIPTION
**Pull request by dairyo**
* Implement onboard for endnode method.
* Move the method from `ThingIFAPI.swift` to `ThingIFAPI+OnBoard.swift`.
* Fix `PendingEndNode` according to thing-if AndroidSDK.
* Add small tests
* Delete old tests.

The number of small tests become smaller than old test because an argument is removed from old one. The argument is `OnboardEndnodeWithGatewayOptions`.

Related PR: #296

### CODE REVIEW

File: ThingIFSDK/ThingIFSDK/Gateway/EndNode.swift [hunk 1]
```diff
@@ -39,7 +39,7 @@ public struct EndNode: TargetThing, Equatable {
 extension EndNode: FromJsonObject {
 
     internal init(_ jsonObject: [String : Any]) throws {
-        guard let thingID = jsonObject["thingID"] as? String,
+        guard let thingID = jsonObject["endNodeThingID"] as? String,
```
Comment (yongpingchen) [inline ThingIFSDK/ThingIFSDK/Gateway/EndNode.swift:42]: Should not change it to `endNodeThingID`. jsonObject should be from onboarding result, where is `thingID`. Please confirm again.

**Review by yongpingchen — CHANGES_REQUESTED** [review]

**Review by naoki-akimoto — APPROVED** [review]
LGTM!

File: ThingIFSDK/ThingIFSDK/Gateway/EndNode.swift (hunk 1 above)
Comment (naoki-akimoto) [reply ThingIFSDK/ThingIFSDK/Gateway/EndNode.swift:42]: Android側も"endNodeThingID"なんだけど、Android側を変更した方が良い？

https://github.com/KiiPlatform/thing-if-AndroidSDK/blob/develop/thingif/src/main/java/com/kii/thingif/ThingIFAPI.java#L720

Comment (dairyo) [reply ThingIFSDK/ThingIFSDK/Gateway/EndNode.swift:42]: @yongpingchen 

I wrote this according to implementation of thing-if AndroidSDK. The code is [this](https://github.com/KiiPlatform/thing-if-AndroidSDK/blob/develop/thingif/src/main/java/com/kii/thingif/ThingIFAPI.java#L720)

Is implementation of thing-if AndroidSDK is also wrong?

Moreover, in old small test, mock response of the thing id is `endNodeThingID`.
I also check specification of gateway onboarding. The sample of response describe as `endNodeThingID`. The specification is [this](https://docs.google.com/document/edit?hgd=1&id=1xNSWBeX_u2DoiIDdFvwRqP_4FQQONQnJyRQcvgUayNs#heading=h.m7ol71wbdhic)

What do you think?

Comment (yongpingchen) [reply ThingIFSDK/ThingIFSDK/Gateway/EndNode.swift:42]: @naoki-akimoto @shinsuke-nara 
Sorry. That is my mistake. You are right. Then LGTM

**Review by yongpingchen — APPROVED** [review]

**Pull request merged**